src/transformers/modeling_reformer.py

```diff
@@ -171,15 +202,23 @@ def forward(self, position_ids):
             )
 
             # compute how many columns are needed
-            required_pos_encodings_columns = -(-sequence_length // self.axial_pos_shape[1])
+            max_position_id = position_ids.max().item()
+            required_pos_encodings_columns = -(-(max_position_id + 1) // self.axial_pos_shape[1])
 
             # cut to columns that are needed
             position_encodings = torch.cat(
                 [weight[:, :required_pos_encodings_columns] for weight in broadcasted_weights], dim=-1
             )
-            position_encodings = torch.reshape(position_encodings, (batch_size, -1, position_encodings.shape[-1]))[
-                :, :sequence_length
-            ]
+            position_encodings = torch.reshape(position_encodings, (batch_size, -1, position_encodings.shape[-1]))
+
+            # select correct position encodings
+            position_encodings = torch.cat(
```

Comment:
can you batch this somehow
`position_encodings.index_select(1, position_ids)` maybe?

Reply:
tried it - doesn't work sadly :-/